element_calcium_imaging/imaging_withpostprocessing.py

```diff
@@ -0,0 +1,1217 @@
+        :param linking_module: a module name or a module containing the
+         required dependencies to activate the `imaging` module:
+         + all that are required by the `scan` module
+    """
+
+    if isinstance(linking_module, str):
+        linking_module = importlib.import_module(linking_module)
+    assert inspect.ismodule(
+        linking_module
+    ), "The argument 'dependency' must be a module's name or a module"
+
+    global _linking_module
+    _linking_module = linking_module
+
+    scan.activate(
+        scan_schema_name,
+        create_schema=create_schema,
+        create_tables=create_tables,
+        linking_module=linking_module,
+    )
+    schema.activate(
+        imaging_schema_name,
+        create_schema=create_schema,
+        create_tables=create_tables,
+        add_objects=_linking_module.__dict__,
+    )
+
+
+# -------------- Table declarations --------------
+
+
+@schema
+class ProcessingMethod(dj.Lookup):
+    definition = """
+    # Method, package, analysis suite used for processing of calcium imaging 
+    # data (e.g. Suite2p, CaImAn, etc.)
+    processing_method: char(8)
+    ---
+    processing_method_desc: varchar(1000)
+    """
+
+    contents = [
+        ("suite2p", "suite2p analysis suite"),
+        ("caiman", "caiman analysis suite"),
+    ]
+
+
+@schema
+class ProcessingParamSet(dj.Lookup):
+    definition = """  #  Parameter set used for processing of calcium imaging data
+    paramset_idx:  smallint
+    ---
+    -> ProcessingMethod
+    paramset_desc: varchar(128)
+    param_set_hash: uuid
+    unique index (param_set_hash)
+    params: longblob  # dictionary of all applicable parameters
+    """
+
+    @classmethod
+    def insert_new_params(
+        cls, processing_method: str, paramset_idx: int, paramset_desc: str, params: dict
+    ):
+        param_dict = {
+            "processing_method": processing_method,
+            "paramset_idx": paramset_idx,
+            "paramset_desc": paramset_desc,
+            "params": params,
+            "param_set_hash": dict_to_uuid(params),
+        }
+        q_param = cls & {"param_set_hash": param_dict["param_set_hash"]}
+
+        if q_param:  # If the specified param-set already exists
+            pname = q_param.fetch1("paramset_idx")
+            if pname == paramset_idx:  # If the existed set has the same name: job done
+                return
+            else:  # If not same name: human error, trying to add the same paramset with different name
+                raise dj.DataJointError(
+                    "The specified param-set already exists - name: {}".format(pname)
+                )
+        else:
+            cls.insert1(param_dict)
+
+
+@schema
+class CellCompartment(dj.Lookup):
+    definition = """  # Cell compartments that can be imaged
+    cell_compartment         : char(16)
+    """
+
+    contents = zip(["axon", "soma", "bouton"])
```

Comment:
There may be `dendrite` or `neurite` (more generally)

Reply:
Which one should we add?

Reply:
Thanks @dimitri-yatsenko.  We can update this lookup table once a user requests further compartments.